### PR TITLE
HAWQ-1052. Fixed extraction logic for boolean datatype.

### PR DESCRIPTION
**Pull request by sansanichfb**

### CODE REVIEW

File: src/backend/access/external/pxfheaders.c [hunk 1]
```diff
@@ -66,8 +66,13 @@ void build_http_header(PxfInputData *input)
 	if (proj_info != NULL && proj_info->pi_isVarList)
 	{
 		List* qualsAttributes = extractPxfAttributes(input->quals);
-
-		add_projection_desc_httpheader(headers, proj_info, qualsAttributes);
+		/* projection information is incomplete if columns from WHERE clause wasn't extracted */
+		if (qualsAttributes !=  NIL || list_length(input->quals) == 0)
```
Comment (kavinderd) [inline src/backend/access/external/pxfheaders.c]: Why is the check for `list_length(input->quals)` with `0`?

**Comment by kavinderd** [comment]
Aside from my confusion with my comment, +1

File: src/backend/access/external/pxfheaders.c (hunk 1 above)
Comment (sansanichfb) [inline src/backend/access/external/pxfheaders.c]: extractPxfAttributes function might return NIL in two cases - when input quals are empty or some error occurred during extraction. So this condition list_length(input->quals) == 0 means no input quals has been passed thus nothing was extracted and it's safe to add projection info.

Comment (kavinderd) [inline src/backend/access/external/pxfheaders.c]: Ok, got it

**Pull request closed without merging**